packages/analytics-browser/src/config.ts

```diff
@@ -62,105 +53,145 @@ export class BrowserConfig extends Config implements IBrowserConfig {
+    this.sessionId = options?.sessionId;
+    this.userId = options?.userId;
 
     this.appVersion = options?.appVersion;
     this.attribution = options?.attribution;
     this.cookieExpiration = options?.cookieExpiration ?? defaultConfig.cookieExpiration;
     this.cookieSameSite = options?.cookieSameSite ?? defaultConfig.cookieSameSite;
     this.cookieSecure = options?.cookieSecure ?? defaultConfig.cookieSecure;
     this.cookieUpgrade = options?.cookieUpgrade ?? defaultConfig.cookieUpgrade;
-    this.deviceId = options?.deviceId;
     this.disableCookies = options?.disableCookies ?? defaultConfig.disableCookies;
     this.domain = options?.domain ?? defaultConfig.domain;
-    this.lastEventTime = this.lastEventTime ?? options?.lastEventTime;
-    this.optOut = Boolean(options?.optOut);
     this.partnerId = options?.partnerId;
-    this.sessionId = options?.sessionId;
+    this.sessionTimeout = options?.sessionTimeout ?? defaultConfig.sessionTimeout;
     this.trackingOptions = options?.trackingOptions ?? defaultConfig.trackingOptions;
-    this.userId = userId;
   }
 
   get deviceId() {
-    return this.sessionManager.getDeviceId();
+    return this._deviceId;
   }
 
   set deviceId(deviceId: string | undefined) {
-    this.sessionManager.setDeviceId(deviceId);
+    if (this._deviceId !== deviceId) {
+      this._deviceId = deviceId;
+      this.updateStorage();
+    }
   }
 
   get userId() {
-    return this.sessionManager.getUserId();
+    return this._userId;
   }
 
   set userId(userId: string | undefined) {
-    this.sessionManager.setUserId(userId);
+    if (this._userId !== userId) {
+      this._userId = userId;
+      this.updateStorage();
+    }
   }
 
   get sessionId() {
-    return this.sessionManager.getSessionId();
+    return this._sessionId;
   }
 
   set sessionId(sessionId: number | undefined) {
-    this.sessionManager.setSessionId(sessionId);
+    if (this._sessionId !== sessionId) {
+      this._sessionId = sessionId;
+      this.updateStorage();
+    }
   }
 
   get optOut() {
-    return this.sessionManager.getOptOut();
+    return this._optOut;
   }
 
   set optOut(optOut: boolean) {
-    this.sessionManager?.setOptOut(Boolean(optOut));
+    if (this._optOut !== optOut) {
+      this._optOut = optOut;
+      this.updateStorage();
+    }
   }
 
   get lastEventTime() {
-    return this.sessionManager.getLastEventTime();
+    return this._lastEventTime;
   }
 
   set lastEventTime(lastEventTime: number | undefined) {
-    this.sessionManager.setLastEventTime(lastEventTime);
+    if (this._lastEventTime !== lastEventTime) {
+      this._lastEventTime = lastEventTime;
+      this.updateStorage();
+    }
+  }
+
+  private updateStorage() {
+    const cache = {
+      deviceId: this._deviceId,
+      userId: this._userId,
+      sessionId: this._sessionId,
+      optOut: this._optOut,
+      lastEventTime: this._lastEventTime,
+    };
+    void this.cookieStorage?.set(getCookieName(this.apiKey), cache);
```

Comment:
Yup, updating persistent storage will have some delay. It should be fine since we read from the cache, except during init.

```
userId = 1 // cache has value of 1; storage has value of undefined (potentially)
console.log(userId) // returns 1 (from cache)
userId = 2 // cache has value of 2; storage has value of undefined (potentially)
console.log(userId) // returns 2 (from cache)

// on succeeding clock ticks
// storage updates to 1
// storage updates to 2
```

Reply:
Oh, sorry, I mean the order or async updates, is the persistent storage updates order is guaranteed? Namely:
```js
// on succeeding clock ticks
// storage try to update to 1
// storage try to update to 2
// storage updates to 2 finish
// storage updates to 1 finish
```

Reply:
Given that they perform the same set of operations, they will likely resolve in the same order.

Reply:
Discussed offline, this is very rare case and will likely not happen in real usage. We haven't seen any issue since last q3 (with current logic).